packages/@glimmer/component/addon/index.ts

```diff
@@ -12,11 +12,11 @@ declare module '@ember/component' {
 import { setComponentManager } from '@ember/component';
 
 import GlimmerComponentManager from './-private/ember-component-manager';
-import _GlimmerComponent from './-private/component';
+import _GlimmerComponent, { Args } from './-private/component';
 import { setOwner } from '@ember/application';
 
-export default class GlimmerComponent extends _GlimmerComponent {
-  constructor(owner, args) {
+export default class GlimmerComponent<S> extends _GlimmerComponent<S> {
+  constructor(owner: object, args: Args<S>) {
```

Comment:
Why is `owner` `object` here and not `unknown` as it is in the parent?

Reply:
A good question, and I had the same one. It's because `setOwner` takes an `object` argument. It’s very likely something we want to rationalize throughout… but not in *this* PR.

Reply:
We've [taught people](https://docs.ember-cli-typescript.com/ember/components#adding-arguments) to do:
```ts
export default class MyComponent extends Component<Args> {
  constructor(owner: unknown, args: Args) {
    super(owner, args);
    ⋮
  }
}
```
So making this `object` here will force users update their constructor type signatures (otherwise the `super()` call doesn't type check: `Argument of type 'unknown' is not assignable to parameter of type 'object'.`).

Reply:
The bit with `unknown` is actually broken today, so we need to update our docs—we have (somewhere!) been publishing with requiring `object` *anyway*, so folks have been needing to introduce dummy `Owner` object blobs.